OpenNefia.Content/UI/Element/FeatWindow.cs

```diff
@@ -0,0 +1,311 @@
+                public bool IsRace;
+                
+                public GainedFeat(FeatPrototype Prototype, int Level, bool isRace) : base(Prototype, Level)
+                {
+                    IsRace = isRace;
+                }
+            }
+        }
+
+        public class FeatCell : UiListCell<FeatData>
+        {
+            private IAssetInstance FeatIcons;
+            public static explicit operator FeatCell(FeatData data) => new FeatCell(data);
+
+            private IUiText DescriptionText;
+            public FeatCell(FeatData data) 
+                : base(data, new UiText())
+            {
+                FeatIcons = Assets.Get(Protos.Asset.FeatIcons);
+                Text = data.Name;
+                DescriptionText = new UiText(Data.Description);
+                UiText.Color = data.Color;
+                DescriptionText.Color = data.Color;
+            }
+
+            private int Offset => Data switch
+            {
+                FeatData.FeatHeader header => 35,
+                FeatData.GainedFeat gained => -8,
+                _ => 0,
+            };
+
+            public override void SetPosition(int x, int y)
+            {
+                base.SetPosition(x + Offset, y);
+                UiText.SetPosition(UiText.X, UiText.Y + 4);
+                DescriptionText.SetPosition(UiText.X + 200, UiText.Y);
+            }
+
+            //overriden because position needs to be slightly altered
+            public override void DrawHighlight()
+            {
+                var width = Math.Clamp(UiText.TextWidth + AssetSelectKey.Width + 8 + XOffset, 10, 480);
+                Graphics.SetBlendMode(BlendMode.Subtract);
+                GraphicsEx.SetColor(ColorSelectedSub);
+                Graphics.Rectangle(DrawMode.Fill, UiText.X - XOffset - 4, UiText.Y - 4, width, 19);
+                Graphics.SetBlendMode(BlendMode.Add);
+                GraphicsEx.SetColor(ColorSelectedAdd);
+                Graphics.Rectangle(DrawMode.Fill, UiText.X - XOffset - 3, UiText.Y - 3, width - 2, 17);
+                Graphics.SetBlendMode(BlendMode.Alpha);
+                GraphicsEx.SetColor(Love.Color.White);
+                AssetListBullet.Draw(UiText.X - XOffset - 5 + width - 20, UiText.Y - 0);
+            }
+
+
+            public override void Draw()
+            {
+                if (IndexInList % 2 == 0 && Data is not FeatData.FeatHeader)
+                {
+                    Love.Graphics.SetColor(UiColors.ListEntryAccent);
+                    Love.Graphics.Rectangle(Love.DrawMode.Fill, X - 1, Y, 650 - Offset, 18);
+                }
+                switch (Data)
+                {
+                    case FeatData.GainedFeat feat:
+                        GraphicsEx.SetColor(Core.Maths.Color.White);
+                        FeatIcons.DrawRegion($"{(int)feat.Prototype.FeatType}", X - 7, Y - 5);
```

Comment:
Maybe have an enum or helper function for this. See `InventoryIcons`. Then you'd convert the `FeatType` into a region ID based on its value.

Reply:
changed in b1407d8